src/udev/udevadm-trigger.c

```diff
@@ -25,6 +25,30 @@ static bool arg_verbose = false;
 static bool arg_dry_run = false;
 static bool arg_quiet = false;
 static bool arg_uuid = false;
+static bool arg_entry_match = false;
+static bool arg_entry_nomatch = false;
+
+static int db_present(sd_device *device) {
+        struct stat statbuf;
+        const char *id, *path;
+        int r;
+
+        assert(device);
+
+        if (sd_device_get_is_initialized(device))
+                return 0;
+
+        r = device_get_device_id(device, &id);
+        if (r < 0)
+                return r;
+
+        path = strjoina("/run/udev/data/", id);
+
+        if (stat(path, &statbuf) != 0)
+                return errno == ENOENT ? ENOENT : -errno;
```

Comment:
so first of all, if you want to check for existance of a file use access(F_OK). But more importantly: the sd_device_get_is_initialized() call already checks exactly the same thing: whether a device has a db entry or not. this test here is hence redundant